engine-tests/src/main/java/org/terasology/unittest/stubs/ListOfObjectComponent.java

```diff
@@ -3,16 +3,28 @@
 
 package org.terasology.unittest.stubs;
 
-import org.terasology.engine.entitySystem.Component;
+import org.terasology.gestalt.entitysystem.component.Component;
 import org.terasology.reflection.MappedContainer;
 
 import java.util.ArrayList;
 import java.util.List;
+import java.util.stream.Collectors;
 
-public class ListOfObjectComponent implements Component {
+public class ListOfObjectComponent implements Component<ListOfObjectComponent> {
     public String shortName;
     public List<SubElement> elements = new ArrayList<>();
 
+    @Override
+    public void copy(ListOfObjectComponent other) {
+        this.shortName = other.shortName;
+        this.elements = other.elements.stream().map((subElement -> {
+            SubElement newSub = new SubElement();
+            newSub.id = subElement.id;
+            newSub.type = subElement.type;
+            return newSub;
+        })).collect(Collectors.toList());
```

Comment:
To directly use best-practices in the tests I'd add a copy constructor for `SubElement`.